includes/settings/tools.php

```diff
@@ -197,32 +208,41 @@ public function __construct() {
 
 	private function get_rollback_versions() {
 		$rollback_versions = get_transient( 'elementor_rollback_versions_' . ELEMENTOR_VERSION );
+
 		if ( false === $rollback_versions ) {
 			$max_versions = 30;
 
-			require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+			$versions = apply_filters( 'elementor/settings/rollback/versions', [] );
 
-			$plugin_information = plugins_api(
-				'plugin_information', [
-					'slug' => 'elementor',
-				]
-			);
+			if ( empty( $versions ) ) {
+				require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
 
-			if ( empty( $plugin_information->versions ) || ! is_array( $plugin_information->versions ) ) {
-				return [];
-			}
+				$plugin_information = plugins_api(
+					'plugin_information', [
+						'slug' => 'elementor',
+					]
+				);
+
+				if ( empty( $plugin_information->versions ) || ! is_array( $plugin_information->versions ) ) {
+					return [];
+				}
 
-			uksort( $plugin_information->versions, 'version_compare' );
-			$plugin_information->versions = array_reverse( $plugin_information->versions );
+				uksort( $plugin_information->versions, 'version_compare' );
```

Comment:
Are you bringing the versions in the right order?

Reply:
YES